Create a standalone i18n-calypso-cli package

jsnajdr opened the pull request:
This PR removes the CLI tool from `i18n-calypso` package and moves it to a new `i18n-calypso-cli`. Technically, they have very little in common: one is a runtime dependency, the other one is a build-time tool. Their NPM dependencies are very different. One has a build step and ships transpiled files, the other one is a Node module and ships as-is.

I also rearranged the `i18n-calypso-cli` files so that the translation extraction tool can be used as a library, too (in addition to the CLI command):
```js
const i18nCLI = require( 'i18n-calypso-cli' );
i18nCLI( { ...options } );
```

That should fix issues found in Automattic/jetpack#13705

**How to test:**
Verify that things like `npm run translate` continue to work. And that that Jetpack issue is really fixed by this.

jsnajdr committed:
i18n-calypso-cli: fix ESLint warning about unnecessary regexp escape

jsnajdr committed:
i18n-calypso-cli: remove duplicate test file

jsnajdr committed:
i18n-calypso-cli: rename the bin script to cli.js

jsnajdr committed:
i18n-calypso-cli: rename i18n.js to index.js to make it easier importable

jsnajdr committed:
i18n-calypso-cli: create a standalone package

jsnajdr committed:
i18n-calypso: add missing events dependency

jsnajdr committed:
Add i18n-calypso-cli to package.json and update lockfile

jsnajdr committed:
Update package versions and changelogs

simison commented on packages/i18n-calypso-cli/package.json:
```diff
@@ -0,0 +1,27 @@
+		"commander": "^2.9.0",
+		"debug": "^3.2.6",
+		"globby": "^9.0.0",
+		"jed": "1.1.1",
```
Any reason to leave `^` (or `~`?) out for `jed`?

jsnajdr replied:
Thanks for spotting this! `i18n-calypso-cli` actually doesn't need that dependency. Removed in 30923e0 🔪

akirk reviewed (approved):
I think it's a good call to separate this out but I don't think I have enough context about the Jetpack problem. So since this change has its own merrit anyway, it is good to go for me.

simison commented on packages/i18n-calypso/package.json:
```diff
@@ -13,25 +13,20 @@
 	"author": "@automattic",
 	"license": "GPL-2.0-or-later",
-	"bin": {
-		"i18n-calypso": "cli/index.js"
```
In the readme file? :-)

I don't think there's technical justification to add to `optionalDependencies`/`peerDependencies` since the package really doesn't depend on the CLI?

akirk replied:
Well, it's the only tool that can extract translatable strings wrapped in `translate()` calls, no?

ockham replied:
Still not a super-hard dep IMO. Like it's not required for `i18n-calypso` to work on a readily string-extracted source.

jsnajdr replied:
`package.json` dependencies are not for humans, but for machines to figure out the NPM dependency tree. But we can add a section in README.

ockham commented on packages/i18n-calypso-cli/cli.js:
```diff
@@ -91,7 +91,7 @@ if ( linesFile ) {
 	for ( const line in lines ) {
 		lines[ line ] = lines[ line ].map( String );
-		const modPath = path.relative( __dirname, line ).replace( /^[\/.]+/, '' );
+		const modPath = path.relative( __dirname, line ).replace( /^[/.]+/, '' );
```
`prettier` tends to remove these automatically for me. I meant to look it up at some point, but didn't actually ever do it. Then again, it's `prettier`! It _has_ to know better than I do :laughing:

jsnajdr replied:
Prettier doesn't remove these, but there is a rather recent ESLint rule that warns about unneeded escapes. The JS parser seems to smart enough to detect the `[]` group. It has a [special _RegularExpressionClass_ rule in the JS grammar](https://www.ecma-international.org/ecma-262/5.1/#sec-7.8.5).

Thanks @akirk for pointing this out! I just mindlessly obeyed ESLint and didn't realize that it's not so obvious that the backslash is not needed 🙂

akirk commented:
Well, see pxLjZ-5wA-p2

jsnajdr commented on packages/i18n-calypso-cli/package.json:
```diff
@@ -0,0 +1,27 @@
+{
+	"name": "i18n-calypso-cli",
+	"version": "4.0.0",
```
My idea was to keep both packages in sync version-wise. But now I realize that they are both fairly independent and not likely to be updated at the same time. I'll start with 1.0.0 instead :+1:

jsnajdr committed:
i18n-calypso-cli: doesn't need the Jed dependency

jsnajdr committed:
i18n-calypso-cli: set package version to 1.0.0

jsnajdr committed:
i18n-calypso-cli: update package name references and paths in README

jsnajdr committed:
i18n-calypso: update README

jsnajdr commented:
Addressed @akirk's review comments:
- changed version of `i18n-calypso-cli` package to 1.0.0 -- starting from start
- added mention of `i18n-calypso-cli` to the README of `i18n-calypso`
- fixed package name references and paths in  the `i18n-calypso-cli` README

jsnajdr committed:
i18n-calypso-cli: fixed ESLint warnings (imports, regex escape)
